### PR TITLE
Improve docs for FSharp.Core

### PR DESCRIPTION
**Pull request by dsyme**
This uses three features of `fsdocs` to improve the doc content for FSharp.Core.xml.

This give much better documentation for FSharp.Core, e.g. at https://github.com/fsharp/fsharp-core-docs and https://fsharp.github.io/fsharp-core-docs/

The features are:

1. ordered categories `<category>` and `<categoryindex>`

2. exclusion using `<exclude />`  (note, the same extension to XML docs is used by sandcastle)

3. documentation for namespaces `<namespacesummary>` and `<namespaceremarks>`

See https://github.com/fsprojects/FSharp.Formatting/pull/580 where the support for these features is added

Note that the C# XML doc specification explicitly says it's ok to use extensions to XML docs to support other documentation generation systems. And that is of course the "X" in "XML".  From https://docs.microsoft.com/en-us/dotnet/csharp/codedoc:

> All the tags outlined above represent those that are recognized by the C# compiler. However, a user is free to define their own tags. Tools like Sandcastle bring support for extra tags like <event> and <note>, and even support documenting namespaces. Custom or in-house documentation generation tools can also be used with the standard tags and multiple output formats from HTML to PDF can be supported.

Given the importance of good quality FSharp.Core doc generation I think it's reasonable for us to add this extra information into the FSharp.Core XML docs.

### CODE REVIEW

**Comment by cartermp** [comment]
I don't want to move forward on any of this until we can come to a consensus on these kinds of PRs. The more this branch moves and master does not, the harder it will be to actually incorporate it (for example - for F# 5). I would much rather that these changes just go into master and flow through our normal shipping channels so that when an FSharp.Core 5.0 is cut it will simply have these improvements

**Comment by dsyme** [comment]
> I don't want to move forward on any of this until we can come to a consensus on these kinds of PRs. The more this branch moves and master does not

So far @KevinRansom has incorporated the changes into master almost immediately.  I don't see the problem at all with this.

**Comment by dsyme** [comment]
(To be clear, feature/fsdocs is currently identical to master, with the exception of this PR.)

**Comment by dsyme** [comment]
> I don't want to move forward on any of this until we can come to a consensus on these kinds of PRs. The more this branch moves and master does not...

I re-read this comment and there's a mis-understanding here.  This PR pushes doc changes to master.

**Comment by dsyme** [comment]
> I don't want to move forward on any of this until we can come to a consensus on these kinds of PRs. The more this branch moves and master does not...

I re-read this comment and there's a mis-understanding here.  This PR pushes doc changes to master.

**Comment by cartermp** [comment]
Sorry, I wasn't paying attention to the target branch here. I'm highly allergic to more of these branches sitting around, especially with the added bot traffic. It creates so much noise that anyone other than MS employees are strongly discouraged from following the repo

**Comment by dsyme** [comment]
> Sorry, I wasn't paying attention to the target branch here. I'm highly allergic to more of these branches sitting around, especially with the added bot traffic. It creates so much noise that anyone other than MS employees are strongly discouraged from following the repo

Yes, I can see the problem.  I'd perfectly fine with having feature branches out of repo if only we could easily get best CI and (for docs) automatic integration PRs set up across repos.  

I've added https://github.com/fsharp/fslang-design/blob/master/tooling/FST-1031-xmldoc-extensions.md to document the tags used here, which I'd like you to double check please.  In the process of writing the RFC I adjusted the tags, I'll update the FSharp.Formatting implementation and this PR now to correspond.

**Comment by abelbraaksma** [comment]
> It creates so much noise that anyone other than MS employees are strongly discouraged from following the repo

It indeed clutters the notifications, and I click a lot of it away each day (not such a big deal, but still). There's currently no way to tell github that I'm not interested in bot updates. Maybe this could be requested to the github team? (unf they don't have a public repo themselves, nor a public bug tracking system afaik).

**Comment by auduchinok** [comment]
> I'd perfectly fine with having feature branches out of repo if only we could easily get best CI and (for docs) automatic integration PRs set up across repos.

This would be really great!

**Comment by cartermp** [comment]
> Maybe this could be requested to the github team? 

If only they responded to feature requests coming internally from MS, we wouldn't have had this issue for a long time now :(

>  I'd perfectly fine with having feature branches out of repo if only we could easily get best CI and (for docs) automatic integration PRs set up across repos.

If it requires regularly syncing back into this one then it's just rearranging where the bot PRs come from. If it's only in one direction - dotnet/fsharp --> something/something - then that's probably fine.

**Comment by dsyme** [comment]
Thanks for merging this, much appreciated

File: src/fsharp/FSharp.Core/result.fsi [hunk 1]
```diff
@@ -5,7 +5,9 @@ namespace Microsoft.FSharp.Core
     open Microsoft.FSharp.Core.LanguagePrimitives.IntrinsicOperators
 
     [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
-    /// <summary>Basic operations for values of type <see cref="T:Microsoft.FSharp.Core.Result`2"/>.</summary>
+    /// <summary>Contains operations for working with values of type <see cref="T:Microsoft.FSharp.Core.Result`2"/>.</summary>
```
Comment (cartermp) [inline src/fsharp/FSharp.Core/result.fsi:8]: Note that this and others leads to a 404 in the docs page

**Pull request merged**